memparse.go

```diff
@@ -0,0 +1,202 @@
+	traverseTree = func(root *crit.PsTree, checkpointOutputDir string) error {
+		memReader, err := crit.NewMemoryReader(
+			filepath.Join(checkpointOutputDir, metadata.CheckpointDirectory),
+			root.PID, pageSize,
+		)
+		if err != nil {
+			return err
+		}
+
+		pagemapEntries := memReader.GetPagemapEntries()
+
+		var memSize int64
+
+		for _, entry := range pagemapEntries {
+			memSize += int64(*entry.NrPages) * int64(pageSize)
+		}
+
+		table.Append([]string{
+			fmt.Sprintf("%d", root.PID),
+			root.Comm,
+			metadata.ByteToString(memSize),
+		})
+
+		for _, child := range root.Children {
+			if err := traverseTree(child, checkpointOutputDir); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	for _, task := range tasks {
+		// Clear the table before processing each checkpoint task
+		table.ClearRows()
+
+		c := crit.New(nil, nil, filepath.Join(task.outputDir, "checkpoint"), false, false)
+		psTree, err := c.ExplorePs()
+		if err != nil {
+			return fmt.Errorf("failed to get process tree: %w", err)
+		}
+
+		// Populate the table rows
+		if err := traverseTree(psTree, task.outputDir); err != nil {
+			return err
+		}
+
+		fmt.Printf("\nDisplaying processes memory sizes from %s\n\n", task.checkpointFilePath)
+		table.Render()
+	}
+
+	return nil
+}
+
+func printProcessMemoryPages(task task) error {
+	c := crit.New(nil, nil, filepath.Join(task.outputDir, metadata.CheckpointDirectory), false, false)
+	psTree, err := c.ExplorePs()
+	if err != nil {
+		return fmt.Errorf("failed to get process tree: %w", err)
+	}
+
+	// Check if PID exist within the checkpoint
+	if pID != 0 {
+		ps := findPsTreeByPID(psTree, pID)
+		if ps == nil {
+			return fmt.Errorf("no process with PID %d (use `inspect --ps-tree` to view all PIDs)", pID)
+		}
+	}
+
+	memReader, err := crit.NewMemoryReader(
+		filepath.Join(task.outputDir, metadata.CheckpointDirectory),
+		pID, pageSize,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Unpack pages-[pagesID].img file for the given PID
+	if err := untarFiles(
+		task.checkpointFilePath, task.outputDir,
+		[]string{filepath.Join(metadata.CheckpointDirectory, fmt.Sprintf("pages-%d.img", memReader.GetPagesID()))},
+	); err != nil {
+		return err
+	}
+
+	// Write the output to stdout by default
+	var output io.Writer = os.Stdout
+	var compact bool
+
+	if outputFilePath != "" {
+		// Write to ouput to file if --output is specified
+		f, err := os.Create(outputFilePath)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		output = f
+		fmt.Printf("\nWriting memory pages content for process ID %d from checkpoint: %s to file: %s...\n",
+			pID, task.checkpointFilePath, outputFilePath,
+		)
+	} else {
+		compact = true // Use a compact format when writing the output to stdout
+		fmt.Printf("\nDisplaying memory pages content for process ID %d from checkpoint: %s\n\n", pID, task.checkpointFilePath)
+	}
+
+	fmt.Fprintf(output, "Address           Hexadecimal                                       ASCII            \n")
```

Comment:
It would be better to use `fmt.Fprintln` here, since we're not formatting anything. `fmt.Fprintf` is slower as it needs to check the string for format specifiers and process every argument. Ditto for the line below (143)